routing/router.go

```diff
@@ -1411,6 +1411,59 @@ func (r *ChannelRouter) addZombieEdge(chanID uint64) error {
 	return nil
 }
 
+// makeFundingScript is used to make the funding script for both segwit v0 and
+// segwit v1 (tapoort) channels.
+//
+// TODO(roasbeef: export and use elsewhere?
```

Comment:
yeah I think maybe `input` package would be a good place for this. Then can call that from `ValidateChannel` and other places where we do this switch

Reply:
So the logic here is a bit custom: it relies on that trick I mentioned to set the channel update level feature flag so the router knows to apply this logic for validation. With gossip 1.75, this changes. In other words, most of the logic is only specific to this context, and is temporary.